@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -0,0 +1,175 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { BlobData, OlpClientSettings, TypedEvent } from "@here/olp-sdk-core";
+
+/**
+ * Contains information about the uploaded chunk.
+ */
+export interface UploadStatus {
+    /**
+     * The total count of the uploaded chunks.
+     */
+    uploadedChunks: number;
+    /**
+     * The total count of the chunks.
+     */
+    totalChunks: number;
+    /**
+     * The size of the chunk in bytes.
+     */
+    chunkSize: number;
+    /**
+     * The number of the uploaded chunk.
+     */
+    chunkNumber: number;
+    /**
+     * The ID of the chunk.
+     */
+    chunkId: string;
+}
+
+/**
+ * The options for uploading.
+ */
+export interface UploadOptions {
+    /**
+     * The HRN string of the catalog to upload.
+     */
+    catalogHrn: string;
+
+    /**
+     * The ID of the layer to upload.
+     */
+    layerId: string;
+
+    /**
+     * The version of the blob storage.
+     *
+     * Supports Blob API v1 and v2.
+     */
+    blobVersion: number;
+
+    /**
+     * The data key to use for uploading.
+     *
+     * If you use Blob API v1, the key is the data handle.
+     */
+    handle: string;
+
+    /**
+     * The content type of the object being uploaded.
+     *
+     * Must be equal to the layer content type.
+     */
+    contentType: string;
+
+    /**
+     * Specifies how many parallel requests should be used
+     * for uploading chunks.
+     *
+     * Default value: 6.
+     */
+    parallelRequests?: number;
+
+    /**
+     * The size of each chunk.
+     *
+     * The minimum size is 5 MB, maximum – 5 GB.
+     * The default value is 5 MB.
+     * When an out-of-range value is provided, it is trimmed
+     * to the default value of 5 MB.
+     */
+    chunkSizeMB?: number;
+
+    /**
+     * The content encoding of the object being uploaded.
+     *
+     * Must be equal to the layer content encoding.
+     */
+    contentEncoding?: string;
+
+    /**
+     * An optional free-form tag
+     * used for grouping billing records together.
+     */
+    billingTag?: string;
+
+    /**
+     * In case you are interested in getting feedback,
+     * use this to subscribe to the following events:.
+     *
+     * Use it to subscribe to the following events:
+     * - `onStart`, which starts the `MultiPart Upload` process.
+     * - `onComplete`, which completes the `MultiPart Upload` process.
+     * - `onStatus`, which uploads each chunk.
+     */
+    onStart?: (status: { token: string }) => void;
```

Comment:
```suggestion
    onStart?: (status: { multipartToken: string }) => void;
```

If the token is the multipartToken returned by the startMultipartUploadByKey API then lets name it same as it is named in the REST API.